var/spack/repos/builtin/packages/miniforge3/package.py

```diff
@@ -3,50 +3,63 @@
+        "MacOSX-x86_64": ("25ca082ab00a776db356f9bbc660edf6d24659e2aec1cbec5fd4ce992d4d193d"),
+    },
+}
 
 
 class Miniforge3(Package):
-    """Miniforge3 is a minimal installer for conda specific to conda-forge."""
+    """Miniforge3 is a minimal installer for conda and mamba specific to conda-forge."""
 
     homepage = "https://github.com/conda-forge/miniforge"
-    url = "https://github.com/conda-forge/miniforge/releases/download/4.8.3-2/Miniforge3-4.8.3-2-Linux-x86_64.sh"
-
-    version(
-        "4.8.3-4-Linux-x86_64",
-        url="https://github.com/conda-forge/miniforge/releases/download/4.8.3-4/Miniforge3-4.8.3-4-Linux-x86_64.sh",
-        sha256="24951262a126582f5f2e1cf82c9cd0fa20e936ef3309fdb8397175f29e647646",
-        expand=False,
-    )
-    version(
-        "4.8.3-4-Linux-aarch64",
-        url="https://github.com/conda-forge/miniforge/releases/download/4.8.3-4/Miniforge3-4.8.3-4-Linux-aarch64.sh",
-        sha256="52a8dde14ecfb633800a2de26543a78315058e30f5883701da1ad2f2d5ba9ed8",
-        expand=False,
-    )
-    version(
-        "4.8.3-2-Linux-x86_64",
-        url="https://github.com/conda-forge/miniforge/releases/download/4.8.3-2/Miniforge3-4.8.3-2-Linux-x86_64.sh",
-        sha256="c8e5b894fe91ce0f86e61065d2247346af107f8d53de0ad89ec848701c4ec1f9",
-        expand=False,
-    )
-    version(
-        "4.8.3-2-Linux-aarch64",
-        url="https://github.com/conda-forge/miniforge/releases/download/4.8.3-2/Miniforge3-4.8.3-2-Linux-aarch64.sh",
-        sha256="bfefc0ede6354568978b4198607edd7f17c2f50ca4c6a47e9f22f8c257c8230a",
-        expand=False,
-    )
-    version(
-        "4.8.3-2-MacOSX-x86_64",
-        url="https://github.com/conda-forge/miniforge/releases/download/4.8.3-2/Miniforge3-4.8.3-2-MacOSX-x86_64.sh",
-        sha256="25ca082ab00a776db356f9bbc660edf6d24659e2aec1cbec5fd4ce992d4d193d",
-        expand=False,
-    )
+
+    maintainers("ChristopherChristofi")
+
+    license("BSD-3-Clause")
+
+    variant("mamba", default=True, description="Enable mamba support.")
+
+    conflicts("+mamba", when="@:22.3.1-0")
+
+    for ver, packages in _versions.items():
+        key = "{0}-{1}".format(platform.system(), platform.machine())
+        pkg = packages.get(key)
+        if pkg:
+            version(ver, sha256=pkg[0], expand=False)
+
+    def url_for_version(self, version):
+        url = "https://github.com/conda-forge/miniforge/releases/download/{0}/Miniforge3-{0}-{1}-{2}.sh"
+        return url.format(version, platform.system(), platform.machine())
 
     def install(self, spec, prefix):
-        mkdirp(prefix)
-        pkgname = "Miniforge3-{0}.sh".format(self.version)
-        chmod = which("chmod")
-        chmod("+x", pkgname)
-        sh = which("sh")
-        sh("./{0}".format(pkgname), "-b", "-f", "-s", "-p", prefix)
+        dir, script = split(self.stage.archive_file)
+        bash = which("bash")
+        bash(script, "-b", "-f", "-p", self.prefix)
```

Comment:
Is this process the same for all listed versions or the more recent one(s)?

Reply:
Hello @tldahlgren , thank you for your comments, yes this process works for all older versions - the `-s` option is for skipping the pre/post-link/install scripts. Which is also a available option for both the `anaconda3` and `miniconda3` install scripts but not included in their spack package recipes